Zend/zend_max_execution_timer.c

```diff
@@ -35,6 +35,12 @@
 
 ZEND_API void zend_max_execution_timer_init(void) /* {{{ */
 {
+	pid_t pid = getpid();
```

Comment:
This is an existing issue, but in case of `pcntl_fork`, shouldn't the old (inherited) timer be deleted?

Reply:
I don't think so (see https://github.com/php/php-src/pull/13027#discussion_r1442091789) since the inherited timer would be deleted by the original thread/process.

Reply:
Actually, it's not necessary because:

https://man7.org/linux/man-pages/man2/timer_create.2.html#NOTES

> Timers are not inherited by the child of a [fork(2)](https://man7.org/linux/man-pages/man2/fork.2.html), and are disarmed and deleted during an [execve(2)](https://man7.org/linux/man-pages/man2/execve.2.html).

Reply:
According to the manpage of `timer_create()`, they are not inherited by the child of a fork. The pid check in `zend_max_execution_timer_shutdown()` is used to prevent attempts to delete a `EG(max_execution_timer_timer)` inherited from the parent, as it's not a valid timer id in the child.